### PR TITLE
fix: added optional username field to User table

### PR DESCRIPTION
**Pull request by hamzamaimi**
- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
There was an error about the username during the sign in operation, I fixed it by adding the username field to the schema.prisma as an optional field.
Screenshot 2024-07-24 at 10 00 31 am

I also executed the tests, there is just a failure but it's nothing related about what I've done, It's a missing title... 
Screenshot 2024-07-24 at 10 22 09 am
Screenshot 2024-07-24 at 10 30 43 am

To apply the last migration that I've added make sure to run the db:migrate:dev script.

Thank you.

### CODE REVIEW

**Comment by eddiejaoude** [comment]
Please can you raise an issue first to discuss the problem.

File: prisma/schema.prisma [hunk 1]
```diff
@@ -19,6 +19,7 @@ model User {
   email         String          @unique
   emailVerified DateTime?
   image         String?
+  username      String?
```
Comment (eddiejaoude) [inline prisma/schema.prisma:22]: I think this should be required and also unique

```suggestion
  username         String          @unique
```

File: prisma/schema.prisma [hunk 2]
```diff
@@ -19,6 +19,7 @@ model User {
   email         String          @unique
   emailVerified DateTime?
   image         String?
+  username         String          @unique
```
Comment (eddiejaoude) [inline prisma/schema.prisma:22]: I will fix the formatting, as difficult to do it correctly in an inline comment

Comment (eddiejaoude) [reply prisma/schema.prisma:22]:
```suggestion
username      String          @unique
```

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for raising this bug 👍 

After further testing, currently the username field in the database is populated with the github id not the username. This is because of a bug on this line
https://github.com/EddieHubCommunity/HealthCheck/blob/c1f8efeb78a68f4e9e28303c72427343ad5af568/src/app/api/auth/%5B...nextauth%5D/route.js#L18

**Comment by eddiejaoude** [comment]
We don't need the `username`, so I will fix by removing this line 
https://github.com/EddieHubCommunity/HealthCheck/blob/c1f8efeb78a68f4e9e28303c72427343ad5af568/src/app/api/auth/%5B...nextauth%5D/route.js#L18

Thank you again for rasing, but the migration change is hiding the actual problem, so I will remove the line above which will fix the issue. So I will close this PR. Thank you again

**Comment by eddiejaoude** [comment]
Thank you again for raising this 👍  Here is the fix I made so the migration is not required https://github.com/EddieHubCommunity/HealthCheck/commit/7199955cd542b298d97f20cd21dfde8f9922dffb

**Comment by hamzamaimi** [comment]
Ok, I get it. 
Thank you.

**Pull request closed without merging**